PhpAmqpLib/Connection/AMQPStreamConnection.php

```diff
@@ -45,6 +45,15 @@ public function __construct(
         $ssl_protocol = null,
         ?AMQPConnectionConfig $config = null
     ) {
+        if (func_num_args() === 17) {
```

Comment:
We need this warning only when class is used directly, $ssl_protocol is not empty and $config is null

Reply:
I think we need this warning in the following cases (when there are 17 arguments OR $ssl_protocol presents):
`__construct(...15 arguments..., $ssl_protocol, $config)`
`__construct(...15 arguments..., $ssl_protocol, null)`
`__construct(...15 arguments..., null, null)`
`__construct(...15 arguments..., $ssl_protocol)`
But not in the following cases:
`__construct(...15 arguments..., $config)`
`__construct(...15 arguments..., null)`
`__construct(...15 arguments...)`

Now it should be correct.